Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -15,6 +15,7 @@ private let KVOLoading = "loading"
 private let KVOEstimatedProgress = "estimatedProgress"
 
 private let HomeURL = "about:home"
+private let URLBeforePathRegex = NSRegularExpression(pattern: "^https?://([^/]+/)", options: nil, error: nil)!
```

Comment:
The SVC can implement more than one protocol. I don't think you want to confound these two into one large one. i.e. SVC should implement a URLSelector Protocol (whatever we decided to call it) and a separate AutoCompleteDataSource one (or whatever we call that)